msos/StackWalker.cs

```diff
@@ -0,0 +1,373 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Diagnostics.Runtime;
+using System.Runtime.InteropServices;
+
+using static msos.NativeStructs;
+using static msos.NativeMethods;
+
+namespace msos
+{
+    /// <summary>
+    /// This class is responsible for fetching function parameters.
+    /// it relies on x86 Calling Convention - passing parameters to function on the stack. 
+    /// </summary>
+    class StackWalkerStrategy_x86 : StackWalkerStrategy
+    {
+        public StackWalkerStrategy_x86(ClrRuntime runtime) : base(runtime)
+        {
+            if (_runtime.DataTarget.Architecture != Architecture.X86)
+            {
+                throw new Exception("Unexpected Architecture");
+            }
+        }
+
+        protected override UnifiedBlockingObject GetCriticalSectionBlockingObject(UnifiedStackFrame frame)
+        {
+            UnifiedBlockingObject result = null;
+            var paramz = GetParams(frame, ENTER_CRITICAL_SECTION_FUNCTION_PARAM_COUNT);
+
+            var address = Convert(paramz[0]);
+
+            byte[] buffer = new byte[Marshal.SizeOf(new CRITICAL_SECTION())];
```

Comment:
There's a `Marshal.SizeOf` overload that takes a `Type`, so you don't have to create a new object here.